environment/src/main/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionService.java

```diff
@@ -0,0 +1,78 @@
+package com.sequenceiq.environment.environment.encryption;
+
+import static com.sequenceiq.environment.parameter.dto.ResourceGroupUsagePattern.USE_MULTIPLE;
+
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.CloudPlatformVariant;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.CreatedEncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.EncryptionResourcesCreationRequest;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.parameter.dto.AzureParametersDto;
+import com.sequenceiq.environment.parameter.dto.AzureResourceGroupDto;
+import com.sequenceiq.environment.parameter.dto.ParametersDto;
+import com.sequenceiq.environment.parameter.dto.ResourceGroupUsagePattern;
+
+@Component
+public class EnvironmentEncryptionService {
+
+    private final CredentialToCloudCredentialConverter credentialToCloudCredentialConverter;
+
+    private final CloudPlatformConnectors cloudPlatformConnectors;
+
+    public EnvironmentEncryptionService(CredentialToCloudCredentialConverter credentialToCloudCredentialConverter,
+            CloudPlatformConnectors cloudPlatformConnectors) {
+        this.credentialToCloudCredentialConverter = credentialToCloudCredentialConverter;
+        this.cloudPlatformConnectors = cloudPlatformConnectors;
+    }
+
+    public CreatedEncryptionResources createEncryptionResources(EnvironmentDto environmentDto, Environment environment) {
+        EncryptionResources encryptionResources = getCloudConnector(environment.getCloudPlatform());
+        return encryptionResources.createDiskEncryptionSet(createEncryptionResourcesCreationRequest(environmentDto));
+    }
+
+    private EncryptionResources getCloudConnector(String cloudPlatform) {
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        return Optional.ofNullable(cloudPlatformConnectors.get(cloudPlatformVariant).encryptionResources())
+                .orElseThrow(() -> new NotFoundException("Encryption resources component not found!"));
```

Comment:
Not sure but the general CB `NotFoundException` might not be the best here. I would rather suggest creating a dedicated exception type `EncryptionResourcesNotFoundException` (direct copy-paste of [NetworkConnectorNotFoundException](https://github.com/hortonworks/cloudbreak/blob/master/environment/src/main/java/com/sequenceiq/environment/network/NetworkConnectorNotFoundException.java)) in this package and throwing that here.